converter.py

```diff
@@ -0,0 +1,275 @@
+        Decision Tree (Classification) would be our title.
+
+    References
+    ----------
+    .. [1] `Airbnb knowledge repo
+            <https://github.com/airbnb/knowledge-repo>`_
+    """
+
+    AUTHOR = 'Ethen Liu'
+    DATE_FORMAT = '%Y-%m-%d'
+    REPO_NAME = 'machine-learning'
+    BASE_URL = 'https://github.com/ethen8181/'
+
+    def __init__(self, knowledge_repo_path, inplace):
+        self.inplace = inplace
+        self.knowledge_repo_path = knowledge_repo_path
+
+    def convert(self, path):
+        """
+        Convert the input path's notebook to a knowledge repo. This
+        will add a mandatory raw cell that contains the yaml information
+        needed by the knowledge repo and an additional cell that contains
+        link to the notebook on github.
+
+        Parameters
+        ----------
+        path : str
+            Path that has the '.ipynb' extension.
+
+        Returns
+        -------
+        notebook : dict
+            Updated Jupyter notebook's raw json represented in dictionary format.
+            Ready to be passed to the .add method to add to the knowledge repo.
+        """
+        self.date_created_ = self._date_created(path)
+        self.date_updated_ = self._date_updated(path)
+        self.tags_, self.github_link_ = self._tags_and_github_link(path)
+        with open(path, encoding = 'utf-8') as f:
+            notebook = json.load(f)
+
+        self.title_ = self._title(notebook)
+
+        # prepend the dictionary header to notebook['cells']
+        notebook['cells'] = ([self._construct_header()] +
+                             [self._construct_github_link_cell()] +
+                             notebook['cells'])
+        if not self.inplace:
+            head, ext = os.path.splitext(path)
+            head += '-converted'
+            path = head + ext
+
+        self._path = path
+        return notebook
+
+    def _date_created(self, path):
+        """Grab the date of creation through git log."""
+        cmd = 'git log --diff-filter=A --follow --format=%cd -1 -- {}'.format(path)
+        return self._git_date_cmd(cmd)
+
+    def _date_updated(self, path):
+        """Grab the last date modified through git log."""
+        cmd = 'git log --format=%cd -1 -- {}'.format(path)
+        return self._git_date_cmd(cmd)
+
+    def _git_date_cmd(self, cmd):
+        """Run bash command to retrieve and format date string."""
+        date_str = subprocess.check_output(cmd, shell = True)
+        date_dt = date_parser.parse(date_str)
+        formatted_date = date_dt.strftime(self.DATE_FORMAT)
+        return formatted_date
+
+    def _tags_and_github_link(self, path):
+        """
+        Use file name as tags, e.g. /Users/ethen/machine-learning/trees/decision_tree.ipynb
+        we would use 'decision_tree' as the tag
+        """
+        _, file_path = path.split(self.REPO_NAME)
+        file_name = os.path.split(file_path)[-1]
```

Comment:
Unpack with `_, file_name = os.path.split(file_path)`

Reply:
sure, will change this.